DataFormats/HGCalReco/interface/TICLCandidate.h

```diff
@@ -0,0 +1,68 @@
+#ifndef DataFormats_HGCalReco_TICLCandidate_h
+#define DataFormats_HGCalReco_TICLCandidate_h
+
+#include "DataFormats/Candidate/interface/LeafCandidate.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/HGCalReco/interface/Trackster.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+
+// A TICLCandidate is a lightweight physics object made from one or multiple Tracksters.
+
+namespace ticl {
+  class TICLCandidate : public reco::LeafCandidate {
+  public:
+    typedef Trackster::ParticleType ParticleType;
+
+    TICLCandidate(Charge q, const LorentzVector& p4)
+        : LeafCandidate(q, p4), time_(0.f), timeError_(-1.f), rawEnergy_(0.f) {}
+
+    TICLCandidate() : LeafCandidate(), time_(0.f), timeError_(-1.f), rawEnergy_(0.f) {}
+
+    TICLCandidate(const edm::Ptr<Trackster>& trackster)
+        : LeafCandidate(), time_(0.f), timeError_(-1.f), rawEnergy_(0.f), tracksters_({trackster}) {}
+
+    inline float time() const { return time_; }
+    inline float timeError() const { return timeError_; }
+
+    void setTime(float time) { time_ = time; };
+    void setTimeError(float timeError) { timeError_ = timeError; }
+
+    // inline const reco::TrackRef trackRef() const { return trackRef_; }
+    // void setTrackRef(const reco::TrackRef& trackRef);
+
+    inline const edm::Ptr<reco::Track> trackPtr() const { return trackPtr_; }
+    void setTrackPtr(const edm::Ptr<reco::Track>& trackPtr) { trackPtr_ = trackPtr; }
+
+    inline float rawEnergy() const { return rawEnergy_; }
+    void setRawEnergy(float rawEnergy) { rawEnergy_ = rawEnergy; }
+
+    inline const std::vector<edm::Ptr<Trackster> > tracksters() const { return tracksters_; };
+
+    void setTracksters(const std::vector<edm::Ptr<Trackster> >& tracksters) { tracksters_ = tracksters; }
+    void addTrackster(const edm::Ptr<Trackster>& trackster) { tracksters_.push_back(trackster); }
+    // convenience method to return the ID probability for a certain particle type
+    inline float id_probability(ParticleType type) const {
```

Comment:
since there are other ParticleType enums defined elsewhere in CMSSW, probably best to reference this one explicitly as `ticl::Trackster::ParticleType`

Reply:
I guess that'ss why there is a `typdef` few lines above: `typedef Trackster::ParticleType ParticleType;`

Reply:
Ah, I missed it, thanks.